footprint/src/main/java/com/umc/footprint/src/badge/BadgeService.java

```diff
@@ -258,4 +251,53 @@ public BadgeDateInfo modifyRepBadge(String userId, int badgeIdx) throws BaseExce
             throw new BaseException(DATABASE_ERROR);
         }
     }
+
+    @Transactional
+    public List<Integer> getAcquiredBadgeIdxList(int userIdx, Double totalDistance, Integer totalWalkCount) {
+        // 현재 누적 거리, 횟수로 얻을 수 있는 뱃지 계산 - 획득한 뱃지가 없을 때 어떤 동작을 해야 하지?
+        TotalDistanceStatus distanceStatus = TotalDistanceStatus.getDistanceBadge(totalDistance).orElseThrow();
+        TotalRecordStatus recordStatus = TotalRecordStatus.getWalkCountBadge(totalWalkCount).orElseThrow();
+
+        List<Integer> acquiredBadgeIdxList = new ArrayList<>();
+
+        // 현재 갖고 있는 뱃지 조회
+        List<Integer> beforeSavingWalkBadgeList = userBadgeRepository.findAllByUserIdxAndStatus(userIdx, "ACTIVE")
+                .map(userBadgeList -> userBadgeList.stream()
+                        .map(UserBadge::getBadgeIdx)
+                        .sorted()
+                        .collect(Collectors.toList()))
+                .orElse(Collections.emptyList());
+
+        if (beforeSavingWalkBadgeList.isEmpty()) { // 발자국 스타터 부여
+            acquiredBadgeIdxList.add(1);
+        }
+
+        int originMaxDistanceBadgeIdx = 1; // 원래 갖고 있던 뱃지(2~5)의 가장 큰 값
+        int originMaxRecordBadgeIdx = 1; // 원래 갖고 있던 뱃지(6~8)의 가장 큰 값
+
+        for (Integer originBadgeIdx : beforeSavingWalkBadgeList) {
+            if (originBadgeIdx >= TotalDistanceStatus.minBadgeIdx() && originBadgeIdx <= TotalDistanceStatus.maxBadgeIdx()) {
+                originMaxDistanceBadgeIdx = originBadgeIdx;
+            }
+            if (originBadgeIdx >= TotalRecordStatus.minBadgeIdx() && originBadgeIdx <= TotalRecordStatus.maxBadgeIdx()) {
+                originMaxRecordBadgeIdx = originBadgeIdx;
+            }
+        }
```

Comment:
앞에서 beforeSavingWalkBadgeList을 만들 떄 내림차순으로 정렬하고 
originMaxDistanceBadgeIdx, originMaxRecordBadgeIdx 이게 초기화 되면 
바로 for 문을 나가는건 어떤가요?

```suggestion
        for (Integer originBadgeIdx : beforeSavingWalkBadgeList) {
            if (originBadgeIdx >= TotalDistanceStatus.minBadgeIdx() && originBadgeIdx <= TotalDistanceStatus.maxBadgeIdx()) {
                originMaxDistanceBadgeIdx = originBadgeIdx;
            }
            else if (originBadgeIdx >= TotalRecordStatus.minBadgeIdx() && originBadgeIdx <= TotalRecordStatus.maxBadgeIdx()) {
                originMaxRecordBadgeIdx = originBadgeIdx;
            }
            if (originMaxDistanceBadgeIdx != 1 && originMaxRecordBadgeIdx != 1) break;
        }
```